google-ads/src/main/java/com/google/ads/googleads/lib/GoogleAdsClient.java

```diff
@@ -95,10 +98,17 @@ public static Builder newBuilder() {
     clientBuilder
         .setEndpoint(DEFAULT_ENDPOINT)
         .setTransportChannelProvider(transportChannelProvider);
+    clientBuilder.setEnvironment(System.getenv());
     return clientBuilder;
   }
 
-  /** Returns the credentials for this client. */
+  /**
+   * Returns the credentials for this client.
+   *
+   * <p>This field is marked nullable to enable proper builder behavior. In practice, all requests
+   * to Google Ads API must be authenticated.
```

Comment:
nit: ... and this field can never be null.

(we should probably move away from AutoValue for this and similar reasons at some point.)